ts/features/itwallet/walletInstance/store/reducers/index.ts

```diff
@@ -1,21 +1,31 @@
-import * as O from "fp-ts/lib/Option";
-import { flow } from "fp-ts/lib/function";
 import { PersistConfig, persistReducer } from "redux-persist";
-import { createSelector } from "reselect";
 import { getType } from "typesafe-actions";
 import { Action } from "../../../../../store/actions/types";
-import { GlobalState } from "../../../../../store/reducers/types";
 import itwCreateSecureStorage from "../../../common/store/storages/itwSecureStorage";
-import { isWalletInstanceAttestationValid } from "../../../common/utils/itwAttestationUtils";
 import { itwLifecycleStoresReset } from "../../../lifecycle/store/actions";
-import { itwWalletInstanceAttestationStore } from "../actions";
+import {
+  itwWalletInstanceAttestationStore,
+  itwUpdateWalletInstanceStatus,
+  itwWalletInstanceSetAlertShown
+} from "../actions";
+import { WalletInstanceRevocationReason } from "../../../common/utils/itwTypesUtils";
 
 export type ItwWalletInstanceState = {
   attestation: string | undefined;
+  revocation: {
+    isRevoked: boolean;
+    alertShown: boolean;
+    revocationReason?: WalletInstanceRevocationReason;
+  };
```

Comment:
Could this be simplified as follows?

```suggestion
  revocationReason?: WalletInstanceRevocationReason;
```

The existence of the `revocationReason` (!= undefined) is itself a flag which indicates that
- the WI is revoked
- the alert should be displayed

Once the alert is shown, `revocationReason` returns to `undefined`. Makes sense to you?